smithy-aws-cloudformation/src/main/java/software/amazon/smithy/aws/cloudformation/schema/CfnConfig.java

```diff
@@ -0,0 +1,282 @@
+     *
+     * <p>By default, deprecated members are automatically added to the
+     * {@code deprecatedProperties} schema property.
+     *
+     * @param disableDeprecatedPropertyGeneration True to disable {@code deprecatedProperties}
+     *   generation, false otherwise.
+     */
+    public void setDisableDeprecatedPropertyGeneration(boolean disableDeprecatedPropertyGeneration) {
+        this.disableDeprecatedPropertyGeneration = disableDeprecatedPropertyGeneration;
+    }
+
+    public boolean getDisableCapitalizedProperties() {
+        return disableCapitalizedProperties;
+    }
+
+    /**
+     * Set to true to disable automatically capitalizing names of properties
+     * of Resource Schemas.
+     *
+     * <p>By default, property names of Resource Schemas are capitalized if
+     * no {@code cfnName} trait is applied.
+     *
+     * @param disableCapitalizedProperties True to disable capitalizing property names,
+     *   false otherwise.
+     */
+    public void setDisableCapitalizedProperties(boolean disableCapitalizedProperties) {
+        this.disableCapitalizedProperties = disableCapitalizedProperties;
+    }
+
+    public List<String> getExternalDocs() {
+        return externalDocs;
+    }
+
+    /**
+     * Limits the source of converted "externalDocs" fields to the specified
+     * priority ordered list of names in an externalDocumentation trait.
+     *
+     * <p>This list is case insensitive. By default, this is a list of the
+     * following values: "Documentation Url", "DocumentationUrl", "API Reference",
+     * "User Guide", "Developer Guide", "Reference", and "Guide".
+     *
+     * @param externalDocs External docs to look for and convert, in order.
+     */
+    public void setExternalDocs(List<String> externalDocs) {
+        this.externalDocs = externalDocs;
+    }
+
+    public Map<ShapeId, Map<String, Node>> getJsonAdd() {
+        return jsonAdd;
+    }
+
+    /**
+     * Adds or replaces the JSON value in the generated resource schema
+     * document at the given JSON pointer locations with a different JSON
+     * value.
+     *
+     * <p>The value must be a map where each key is a resource shape ID. The
+     * value is a map where each key is a valid JSON pointer string as defined
+     * in RFC 6901. Each value in the nested map is the JSON value to add or
+     * replace at the given target.
+     *
+     * <p>Values are added using similar semantics of the "add" operation
+     * of JSON Patch, as specified in RFC 6902, with the exception that
+     * adding properties to an undefined object will create nested
+     * objects in the result as needed.
+     *
+     * @param jsonAdd Map of JSON path to values to patch in.
+     */
+    public void setJsonAdd(Map<ShapeId, Map<String, Node>> jsonAdd) {
+        this.jsonAdd = Objects.requireNonNull(jsonAdd);
+    }
+
+    @Override
+    public void setUseJsonName(boolean useJsonName) {
```

Comment:
Yeah, probably better to throw on these instead so that they're not ignored and potentially affect generation if the restriction is lifted down the line.